GA/main.py

```
@@ -0,0 +1,81 @@
+res = minimize(problem,
+               algorithm,
+               termination=('n_evals', 100),
+               seed=1,
+               verbose=True,
+               save_history=True)
+
+print(res)
+
+# Save the results to a shelve file to enble easy access to the results
+# and to avoid the need to re-run the optimization in the future.
+myshelf = shelve.open('results.db', 'n')
```

Ruff (0.8.2)

Use a context manager for opening files

(SIM115)